### PR TITLE
Added serving, improved automated testing, refactored backend

### PR DESCRIPTION
**Pull request by miroenev**
*Description of changes:*

1. Added serving (endpoint & RealTimePredictor) for the best model found by HPO.
2. Removed the need for user input (y/n) prior to HPO launch (requested for automated testing).
3. Refactored backend code so that the multiple code-paths are represented by standalone files in the workflows directory.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by miroenev** [comment]
@aaronmarkham Would much appreciate your thoughts, thanks!

File: hyperparameter_tuning/rapids_bring_your_own/rapids_sagemaker_hpo.ipynb [hunk 1]
```diff
@@ -833,7 +876,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "estimator = sagemaker.estimator.Estimator( **estimator_params  )"
+    "estimator = sagemaker.estimator.Estimator(**estimator_params)"
```
Comment (aaronmarkham) [inline hyperparameter_tuning/rapids_bring_your_own/rapids_sagemaker_hpo.ipynb:879]: This won't work with v2 which is rolling out everywhere right now.
Seeing this:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-36-aee0325cbb5f> in <module>
----> 1 estimator = sagemaker.estimator.Estimator(**estimator_params)

TypeError: __init__() missing 1 required positional argument: 'image_uri'
```

Comment (aaronmarkham) [reply hyperparameter_tuning/rapids_bring_your_own/rapids_sagemaker_hpo.ipynb:879]: Recommend updating your kernel to v2.15 or above and retesting...

**Comment by miroenev** [comment]
@aaronmarkham Thanks Aaron, appreciate your help!

We should be all set with the API update (tested with kernel 2.16.1).
Please review when you get a chance.

**Review by aaronmarkham — DISMISSED** [review]

**Comment by aaronmarkham** [comment]
CI isn't totally reliable right now, but it is picking up some errors like: No module named 'dask_cuda'.
Check the [build logs](https://ssr439688g.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=def18621-189c-4fbd-bdf3-911c5727a38f%2Fbuild.log) and search for "traceback".

**Comment by miroenev** [comment]
@aaronmarkham The issues are being caught in the example notebooks that are meant more as a reference for the interested reader. (e.g., hyperparameter_tuning/rapids_bring_your_own/code/workflows/**notebook_references**/multi_GPU.ipynb )

It seems like CI is trying to test them and getting stuck since it doesn't have a GPU + RAPIDS libraries which we assume will be part of the BYOEstimator that we are building up in the main notebook.

There are possibly a few ways forward, curious on your thoughts
1 -- I can remove the offending notebook examples 
2 -- We can flag them so they are skipped in CI.

Please let me know if you have a preference?
Thanks!
-m

**Comment by miroenev** [comment]
@aaronmarkham Just pushed a version that removes the offending notebooks, hopefully this goes through CI.

We can still point to those notebooks at https://github.com/rapidsai/cloud-ml-examples/tree/main/aws/code/workflows/notebook_references

Thanks!
-m

**Review by aaronmarkham — APPROVED** [review]

**Pull request merged**